### ISSUE TITLE
conv node incorrect output on Snapdragon 835

### ISSUE THREAD

**Issue by peteryang1** [ARM-software/armnn#169]
I created some conv nodes in onnx to test, I found in some nodes, GpuAcc and CpuAcc gets different output.

input | kernel | stride | pad | output | status
-- | -- | -- | -- | -- | --
1x3x231x231 | 45x3x3x3 | 2,2 | 1,1,1,1 | 1x45x116x116 | diff
1x3x231x231 | 45x3x3x3 | 1,1 | 1,1,1,1 | 1x45x231x231 | diff
1x4x231x231 | 44x3x3x3 | 1,1 | 1,1,1,1 | 1x44x231x231 | same
1x4x231x231 | 44x3x3x3 | 2x2 | 1,1,1,1 | 1x44x231x231 | same
1x4x231x231 | 45x3x3x3 | 2x2 | 1,1,1,1 | 1x45x231x231 | same
1x4x231x231 | 45x3x3x3 | 2x2 | 0,0,0,0 | 1x45x115x115 | same
1x3x231x231 | 45x3x3x3 | 2x2 | 0,0,0,0 | 1x45x115x115 | diff

I used the master branch on armnn and acl. hope to get some help!

**Comment by peteryang1** [comment]
I update the model and the output here, all input is set to 0.1:
[armnn-result.zip](https://github.com/ARM-software/armnn/files/3038020/armnn-result.zip)

**Comment by MatthewARM** [comment]
Hi @peteryang1 I can ask someone from the Compute Library team to comment, but is this causing a problem in practice? 

Are the differences small or large? Convolutions are complicated floating point operations, so small differences in implementation can lead to rounding happening at different stages. It doesn't necessarily make one implementation "right" and the other "wrong".

**Comment by peteryang1** [comment]
it's very big difference. Since you can tell from the txt file I updatedthanks!

**Comment by MatthewARM** [comment]
OK thatnks @peteryang1, I'm flagging this up internally, we'll try to figure out why GpuAcc would give very different results in these conditions.

**Comment by peteryang1** [comment]
Hi, @MatthewARM . 
Plus, I found the difference appears on snapdragon soc platforms, In my hand is snapdragon 835. While in device with Mali gpu, the result seems fine. Hope to give you some help.

**Comment by morgolock** [comment]
Hi @peteryang1 

I've been trying to run the tests on the model you have attached but `ExecuteNetwork`  cannot load the weights in the binary model file. 

How do you run the tests? Do you use `ExecuteNetwork`? Could you please share the command you type to run the tests.

**Comment by peteryang1** [comment]
Hi, @morgolock 
Because the onnx parser in your code have some problems (It can not load data from raw_data), so I made some change. 
the change is in line 434 of armnn/OnnxParser:
```C++
    // Copy the value list entries into the destination
    if (!onnxTensor.has_raw_data())
    {
        ::memcpy(tensorData.get(), srcData, tensorInfo.GetNumBytes());
    }
    else
    {
        ::memcpy(
            tensorData.get(), onnxTensor.raw_data().c_str(),
            tensorInfo.GetNumBytes());
    }
```
also,I used ExecuteNetwork apis to run the model.

**Comment by peteryang1** [comment]
Hi @morgolock ,
I didn't use linux to compile the code, instead I built the project through clang and ran it on android devices, I followed the file `BuildGuideAndroidNDK.md`.
And in my hand, I have two types of device, one is a huawei p20pro with mali gpu and another oneplus with Qualcomm Snapdragon 835. On my huawei, I get totally same output, but in my oneplus, I got the wrong output.
I put the output from my oneplus device in the file I updated before, it contains gpuacc, cpuacc and cpuref all three modes.
Hope to provide some help to you.
